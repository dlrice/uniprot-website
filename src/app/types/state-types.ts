import { SearchState } from '../../uniprotkb/state/searchInitialState';
import { ResultsState } from '../../uniprotkb/state/resultsInitialState';
import { EntryState } from '../../uniprotkb/state/entryInitialState';
import { MessagesState } from '../../messages/state/messagesInitialState';
import { SearchAction } from '../../uniprotkb/state/searchReducers';
import { ResultAction } from '../../uniprotkb/state/resultsReducers';
import { MessagesAction } from '../../messages/state/messagesReducers';

export type RootState = {
  query: SearchState;
  results: ResultsState;
  entry: EntryState;
  messages: MessagesState;
};

export type RootAction = SearchAction | ResultAction | MessagesAction;
