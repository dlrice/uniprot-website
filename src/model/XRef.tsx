import React, { Fragment } from 'react';
import { v1 } from 'uuid';
import idx from 'idx';
import { InfoList, ExternalLink } from 'franklin-sites';
import {
  entrySectionDatabaseToDatabaseCategory,
  databaseCategoryToString,
} from '../data/database';
import databaseToDatabaseInfoJson from '../data/databaseToDatabaseInfo.json';
import EntrySectionType from '../model/types/EntrySection';
import {
  Database,
  DatabaseCategory,
  DatabaseToDatabaseInfo,
} from '../model/types/databaseTypes';

type Property = {
  key?: string;
  value?: string;
};

export type DatabaseCrossReference = {
  databaseType?: Database;
  id?: string;
  properties?: [Property];
  isoformId?: string;
};

type Data = {
  databaseCrossReferences?: [DatabaseCrossReference];
  primaryAccession?: string;
};

type XrefData = {
  databaseCategory: string;
  xref: DatabaseCrossReference;
};

type XRefProps = {
  section: EntrySectionType;
  data: Data;
};

type XRefItemProps = {
  xRefEntry: DatabaseCrossReference;
  accession: string;
};

type XRefListProps = {
  accession: string;
  xRefData: XrefData[];
  database: Database;
};

type XRefCategoryInfoListProps = {
  databases: Database[];
  xRefData: XrefData[];
  accession: string;
};

const databaseToDatabaseInfo: DatabaseToDatabaseInfo = databaseToDatabaseInfoJson;

type XRefExternalLinkProps = {
  url: string;
  accession?: string | null | undefined;
  id?: string | null | undefined;
  children: string;
};

export const XRefExternalLink: React.FC<XRefExternalLinkProps> = ({
  url,
  accession,
  id,
  children,
}) => {
  let link = url;
  if (accession) {
    link = link.replace(/%acc/g, accession);
  }
  if (id) {
    link = link.replace(/%value/g, id);
  }
  return <ExternalLink url={link}>{children}</ExternalLink>;
};

export const XRefItem: React.FC<XRefItemProps> = ({ xRefEntry, accession }) => {
  const { databaseType: database, properties: entryProperties, id } = xRefEntry;
  if (!id || !database || !accession || !(database in databaseToDatabaseInfo)) {
    return null;
  }
  const info = databaseToDatabaseInfo[database];
  let properties: string = '';
  if (entryProperties) {
    properties = entryProperties
      .map((property: Property) =>
        property.value && property.value === '-' ? '' : property.value
      )
      .join(' ');
  }
  return (
    <li key={v1()}>
      <XRefExternalLink url={info.uriLink} accession={accession} id={id}>
        {id}
      </XRefExternalLink>
      {properties}
    </li>
  );
};

export const XRefList: React.FC<XRefListProps> = ({
  database,
  xRefData,
  accession,
}) => {
  const nodes = xRefData
    .filter(
      (xRefDatum: XrefData) =>
        idx(xRefDatum, _ => _.xref.databaseType) === database
    )
    .map((xRefDatum: XrefData) => {
      const { xref } = xRefDatum;
      if (!xref) {
        return null;
      }
      return (
        <ul className="no-bullet" key={v1()}>
          <XRefItem xRefEntry={xref} accession={accession} />
        </ul>
      );
    });
  return <Fragment>{nodes}</Fragment>;
};

export const XRefCategoryInfoList: React.FC<XRefCategoryInfoListProps> = ({
  databases,
  xRefData,
  accession,
}) => {
  const infoData = databases.sort().map(database => ({
    title: database,
    content: (
      <XRefList database={database} xRefData={xRefData} accession={accession} />
    ),
  }));

  if (infoData.length === 0) {
    return null;
  }
  return <InfoList infoData={infoData} />;
};

export const XRef: React.FC<XRefProps> = ({ data, section }) => {
  const { databaseCrossReferences, primaryAccession: accession } = data;
  if (!databaseCrossReferences || !accession) {
    return null;
  }
  const foundXrefData: XrefData[] = [];
  const foundDatabaseCategoriesToDatabases: {
    [databaseCategory: string]: { [database: string]: boolean };
  } = {};
  for (const xref of databaseCrossReferences) {
    const { databaseType: database } = xref;
    if (!database) {
      continue;
    }
    const databaseCategory = entrySectionDatabaseToDatabaseCategory.get(
      `${section}_${database}`
    );
    if (!databaseCategory) {
      continue;
    }
    foundDatabaseCategoriesToDatabases[databaseCategory] = {
      ...foundDatabaseCategoriesToDatabases[databaseCategory],
      [database]: true,
    };
    foundXrefData.push({ databaseCategory, xref });
  }
  const nodes = Object.keys(foundDatabaseCategoriesToDatabases)
    .sort()
    .map(foundDatabaseCategory => {
      const databases = Object.keys(
        foundDatabaseCategoriesToDatabases[foundDatabaseCategory]
      ) as Database[];
      const databaseCategoryString = databaseCategoryToString.get(
        foundDatabaseCategory as DatabaseCategory
      );
      const infoListNode = (
        <XRefCategoryInfoList
          databases={databases}
          xRefData={foundXrefData}
          accession={accession}
        />
      );
      if (!infoListNode) {
        return null;
      }
      return (
        <Fragment key={v1()}>
          {databaseCategoryString && <h4>{databaseCategoryString}</h4>}
          {infoListNode}
        </Fragment>
      );
    });
  return <Fragment>{nodes}</Fragment>;
};

export default XRef;
