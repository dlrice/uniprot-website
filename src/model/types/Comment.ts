export enum Comment {
  ACTIVITY_REGULATION = 'ACTIVITY REGULATION',
  ALLERGEN = 'ALLERGEN',
  ALTERNATIVE_PRODUCTS = 'ALTERNATIVE PRODUCTS',
  BIOPHYSICOCHEMICAL_PROPERTIES = 'BIOPHYSICOCHEMICAL PROPERTIES',
  BIOTECHNOLOGY = 'BIOTECHNOLOGY',
  CATALYTIC_ACTIVITY = 'CATALYTIC ACTIVITY',
  CAUTION = 'CAUTION',
  COFACTOR = 'COFACTOR',
  DEVELOPMENTAL_STAGE = 'DEVELOPMENTAL STAGE',
  DISEASE = 'DISEASE',
  DISRUPTION_PHENOTYPE = 'DISRUPTION PHENOTYPE',
  DOMAIN = 'DOMAIN',
  FUNCTION = 'FUNCTION',
  INDUCTION = 'INDUCTION',
  INTERACTION = 'INTERACTION',
  MASS_SPECTROMETRY = 'MASS SPECTROMETRY',
  MISCELLANEOUS = 'MISCELLANEOUS',
  PATHWAY = 'PATHWAY',
  PHARMACEUTICAL = 'PHARMACEUTICAL',
  POLYMORPHISM = 'POLYMORPHISM',
  PTM = 'PTM',
  RNA_EDITING = 'RNA EDITING',
  SEQUENCE_CAUTION = 'SEQUENCE CAUTION',
  SIMILARITY = 'SIMILARITY',
  SUBCELLULAR_LOCATION = 'SUBCELLULAR LOCATION',
  SUBUNIT = 'SUBUNIT',
  TISSUE_SPECIFICITY = 'TISSUE SPECIFICITY',
  TOXIC_DOSE = 'TOXIC DOSE',
  UNKNOWN = 'UNKNOWN',
  WEBRESOURCE = 'WEB RESOURCE',
}

export default Comment;
