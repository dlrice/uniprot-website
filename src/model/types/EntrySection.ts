enum EntrySectionType {
  Expression = 'Expression',
  FamilyAndDomains = 'FamilyAndDomains',
  Function = 'Function',
  Interaction = 'Interaction',
  Miscellaneous = 'Miscellaneous',
  NamesAndTaxonomy = 'NamesAndTaxonomy',
  PathologyOrBioTech = 'PathologyOrBioTech',
  ProteinProcessing = 'ProteinProcessing',
  Sequence = 'Sequence',
  Structure = 'Structure',
  SubCellularLocation = 'SubCellularLocation',
}

export default EntrySectionType;
