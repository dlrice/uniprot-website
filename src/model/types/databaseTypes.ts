export enum DatabaseCategory {
  SEQUENCE = 'SEQUENCE',
  STRUCTURE = 'STRUCTURE',
  INTERACTION = 'INTERACTION',
  CHEMISTRY = 'CHEMISTRY',
  FAMILY = 'FAMILY',
  PTM = 'PTM',
  POLYMORPHISM = 'POLYMORPHISM',
  GEL = 'GEL',
  PROTEOMIC = 'PROTEOMIC',
  PROTOCOL = 'PROTOCOL',
  GENOME = 'GENOME',
  ORGANISM = 'ORGANISM',
  PHYLOGENOMIC = 'PHYLOGENOMIC',
  PATHWAY = 'PATHWAY',
  EXPRESSION = 'EXPRESSION',
  DOMAIN = 'DOMAIN',
  OTHER = 'OTHER',
}

type AttributesItem = {
  name: string;
  xmlTag: string;
  uriLink?: string;
};

type DatabaseInfo = {
  name: string;
  displayName: string;
  category: string;
  uriLink: string;
  attributes?: AttributesItem[];
};

export type DatabaseToDatabaseInfo = { [key in Database]: DatabaseInfo };

export enum Database {
  EMBL = 'EMBL',
  CCDS = 'CCDS',
  PIR = 'PIR',
  RefSeq = 'RefSeq',
  UniGene = 'UniGene',
  PDB = 'PDB',
  PDBsum = 'PDBsum',
  DisProt = 'DisProt',
  ProteinModelPortal = 'ProteinModelPortal',
  SMR = 'SMR',
  BioGrid = 'BioGrid',
  ComplexPortal = 'ComplexPortal',
  CORUM = 'CORUM',
  DIP = 'DIP',
  ELM = 'ELM',
  IntAct = 'IntAct',
  MINT = 'MINT',
  STRING = 'STRING',
  BindingDB = 'BindingDB',
  ChEMBL = 'ChEMBL',
  DrugBank = 'DrugBank',
  GuidetoPHARMACOLOGY = 'GuidetoPHARMACOLOGY',
  SwissLipids = 'SwissLipids',
  Allergome = 'Allergome',
  CAZy = 'CAZy',
  ESTHER = 'ESTHER',
  IMGT_GENE_DB = 'IMGT_GENE-DB',
  MEROPS = 'MEROPS',
  MoonDB = 'MoonDB',
  MoonProt = 'MoonProt',
  mycoCLAP = 'mycoCLAP',
  PeroxiBase = 'PeroxiBase',
  REBASE = 'REBASE',
  TCDB = 'TCDB',
  UniLectin = 'UniLectin',
  CarbonylDB = 'CarbonylDB',
  DEPOD = 'DEPOD',
  GlyConnect = 'GlyConnect',
  iPTMnet = 'iPTMnet',
  PhosphoSitePlus = 'PhosphoSitePlus',
  SwissPalm = 'SwissPalm',
  UniCarbKB = 'UniCarbKB',
  BioMuta = 'BioMuta',
  DMDM = 'DMDM',
  dbSNP = 'dbSNP',
  COMPLUYEAST_2DPAGE = 'COMPLUYEAST-2DPAGE',
  DOSAC_COBS_2DPAGE = 'DOSAC-COBS-2DPAGE',
  OGP = 'OGP',
  REPRODUCTION_2DPAGE = 'REPRODUCTION-2DPAGE',
  SWISS_2DPAGE = 'SWISS-2DPAGE',
  UCD_2DPAGE = 'UCD-2DPAGE',
  World_2DPAGE = 'World-2DPAGE',
  EPD = 'EPD',
  MaxQB = 'MaxQB',
  PaxDb = 'PaxDb',
  PeptideAtlas = 'PeptideAtlas',
  PRIDE = 'PRIDE',
  ProMEX = 'ProMEX',
  ProteomicsDB = 'ProteomicsDB',
  TopDownProteomics = 'TopDownProteomics',
  jPOST = 'jPOST',
  DNASU = 'DNASU',
  Ensembl = 'Ensembl',
  EnsemblBacteria = 'EnsemblBacteria',
  EnsemblFungi = 'EnsemblFungi',
  EnsemblMetazoa = 'EnsemblMetazoa',
  EnsemblPlants = 'EnsemblPlants',
  EnsemblProtists = 'EnsemblProtists',
  GeneDB = 'GeneDB',
  GeneID = 'GeneID',
  Gramene = 'Gramene',
  KEGG = 'KEGG',
  PATRIC = 'PATRIC',
  UCSC = 'UCSC',
  VectorBase = 'VectorBase',
  WBParaSite = 'WBParaSite',
  ArachnoServer = 'ArachnoServer',
  Araport = 'Araport',
  CGD = 'CGD',
  ConoServer = 'ConoServer',
  CTD = 'CTD',
  dictyBase = 'dictyBase',
  DisGeNET = 'DisGeNET',
  EchoBASE = 'EchoBASE',
  EcoGene = 'EcoGene',
  euHCVdb = 'euHCVdb',
  EuPathDB = 'EuPathDB',
  FlyBase = 'FlyBase',
  GeneCards = 'GeneCards',
  GeneReviews = 'GeneReviews',
  H_InvDB = 'H-InvDB',
  HGNC = 'HGNC',
  HPA = 'HPA',
  LegioList = 'LegioList',
  Leproma = 'Leproma',
  MaizeGDB = 'MaizeGDB',
  MalaCards = 'MalaCards',
  MGI = 'MGI',
  MIM = 'MIM',
  neXtProt = 'neXtProt',
  OpenTargets = 'OpenTargets',
  Orphanet = 'Orphanet',
  PharmGKB = 'PharmGKB',
  PomBase = 'PomBase',
  PseudoCAP = 'PseudoCAP',
  RGD = 'RGD',
  SGD = 'SGD',
  TAIR = 'TAIR',
  TubercuList = 'TubercuList',
  VGNC = 'VGNC',
  WormBase = 'WormBase',
  Xenbase = 'Xenbase',
  ZFIN = 'ZFIN',
  eggNOG = 'eggNOG',
  GeneTree = 'GeneTree',
  HOGENOM = 'HOGENOM',
  HOVERGEN = 'HOVERGEN',
  InParanoid = 'InParanoid',
  KO = 'KO',
  OMA = 'OMA',
  OrthoDB = 'OrthoDB',
  PhylomeDB = 'PhylomeDB',
  TreeFam = 'TreeFam',
  BioCyc = 'BioCyc',
  BRENDA = 'BRENDA',
  Reactome = 'Reactome',
  SABIO_RK = 'SABIO-RK',
  SignaLink = 'SignaLink',
  SIGNOR = 'SIGNOR',
  UniPathway = 'UniPathway',
  ChiTaRS = 'ChiTaRS',
  EvolutionaryTrace = 'EvolutionaryTrace',
  GeneWiki = 'GeneWiki',
  GenomeRNAi = 'GenomeRNAi',
  PMAP_CutDB = 'PMAP-CutDB',
  PRO = 'PRO',
  Bgee = 'Bgee',
  CleanEx = 'CleanEx',
  CollecTF = 'CollecTF',
  ExpressionAtlas = 'ExpressionAtlas',
  Genevisible = 'Genevisible',
  CDD = 'CDD',
  Gene3D = 'Gene3D',
  HAMAP = 'HAMAP',
  InterPro = 'InterPro',
  PANTHER = 'PANTHER',
  Pfam = 'Pfam',
  PIRSF = 'PIRSF',
  PRINTS = 'PRINTS',
  ProDom = 'ProDom',
  SFLD = 'SFLD',
  SMART = 'SMART',
  SUPFAM = 'SUPFAM',
  TIGRFAMs = 'TIGRFAMs',
  PROSITE = 'PROSITE',
  GO = 'GO',
  Proteomes = 'Proteomes',
  GlycoProtDB = 'GlycoProtDB',
  ProtClustDB = 'ProtClustDB',
  PathwayCommons = 'PathwayCommons',
}

export default Database;
