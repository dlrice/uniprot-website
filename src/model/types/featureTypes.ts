enum FeatureTypes {
  CHAIN = 'chain',
  PTM = 'Post-translationally modified residue',
  CROSSLINK = 'cross-link',
  ACTIVE_SITE = 'active site',
  OTHER_BINDING_SITE = 'Other binding site',
  NUCLEOTIDE_BINDING = 'Nucleotide-binding region',
}

export default FeatureTypes;
