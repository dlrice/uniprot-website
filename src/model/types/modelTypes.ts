export type EvidenceType = {
  evidenceCode: string; // TODO this needs to be an ECO instead of string
  source: string;
  id: string;
};

export type ValueWihEvidence = {
  value: string;
  evidences: EvidenceType[];
};

export type Property = {
  key?: string;
  value?: string;
};
