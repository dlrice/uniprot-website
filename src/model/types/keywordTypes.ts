import EntrySection from './EntrySection';

export enum KeywordCategory {
  BIOLOGICAL_PROCESS = 'Biological process',
  CELLULAR_COMPONENT = 'Cellular component',
  CODING_SEQUENCE_DIVERSITY = 'Coding sequence diversity',
  DEVELOPMENTAL_STAGE = 'Developmental stage',
  DISEASE = 'Disease',
  DOMAIN = 'Domain',
  LIGAND = 'Ligand',
  MOLECULAR_FUNCTION = 'Molecular function',
  PTM = 'PTM',
  TECHNICAL_TERM = 'Technical term',
}

export const entrySectionToKeywordCategories = new Map<
  EntrySection,
  KeywordCategory[]
>();
entrySectionToKeywordCategories.set(EntrySection.Expression, [
  KeywordCategory.DEVELOPMENTAL_STAGE,
]);
entrySectionToKeywordCategories.set(EntrySection.ProteinProcessing, [
  KeywordCategory.PTM,
]);
entrySectionToKeywordCategories.set(EntrySection.SubCellularLocation, [
  KeywordCategory.CELLULAR_COMPONENT,
]);
entrySectionToKeywordCategories.set(EntrySection.Miscellaneous, [
  KeywordCategory.TECHNICAL_TERM,
]);
entrySectionToKeywordCategories.set(EntrySection.PathologyOrBioTech, [
  KeywordCategory.DISEASE,
]);
entrySectionToKeywordCategories.set(EntrySection.FamilyAndDomains, [
  KeywordCategory.DOMAIN,
]);
entrySectionToKeywordCategories.set(EntrySection.Sequence, [
  KeywordCategory.CODING_SEQUENCE_DIVERSITY,
]);
entrySectionToKeywordCategories.set(EntrySection.Function, [
  KeywordCategory.MOLECULAR_FUNCTION,
  KeywordCategory.BIOLOGICAL_PROCESS,
  KeywordCategory.LIGAND,
]);

export default KeywordCategory;
