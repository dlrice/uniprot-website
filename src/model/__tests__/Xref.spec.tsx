import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { XRef, XRefItem, XRefList, XRefCategoryInfoList } from '../XRef';
import data from './modelData.json';
import EntrySectionType from '../../model/types/EntrySection';
import { Database, DatabaseCategory } from '../types/databaseTypes';

configure({ adapter: new Adapter() });

describe('XRef', () => {
  for (const section in EntrySectionType) {
    test(`should render for ${section} section`, () => {
      const wrapper = shallow(
        <XRef data={data} section={section as EntrySectionType} />
      );
      expect(wrapper).toMatchSnapshot();
    });
  }
});

describe('XRefCategoryInfoList', () => {
  test('should render for database category GENOME and Ensembl', () => {
    const { databaseCrossReferences, primaryAccession } = data;
    const wrapper = shallow(
      <XRefCategoryInfoList
        databases={[Database.Ensembl]}
        xRefData={databaseCrossReferences}
        accession={primaryAccession}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('XRefList', () => {
  test('should render for database Ensembl', () => {
    const { databaseCrossReferences, primaryAccession } = data;
    const wrapper = shallow(
      <XRefList
        key={1}
        database={Database.Ensembl}
        xRefData={databaseCrossReferences}
        accession={primaryAccession}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('XRefItem', () => {
  test('should render', () => {
    const { databaseCrossReferences, primaryAccession } = data;
    const wrapper = shallow(
      <XRefItem
        xRefEntry={databaseCrossReferences[0]}
        accession={primaryAccession}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
