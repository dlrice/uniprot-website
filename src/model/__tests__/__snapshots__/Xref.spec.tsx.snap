// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XRef should render for Expression section 1`] = `"<Fragment />"`;

exports[`XRef should render for FamilyAndDomains section 1`] = `"<Fragment />"`;

exports[`XRef should render for Function section 1`] = `"<Fragment />"`;

exports[`XRef should render for Interaction section 1`] = `"<Fragment />"`;

exports[`XRef should render for Miscellaneous section 1`] = `"<Fragment />"`;

exports[`XRef should render for NamesAndTaxonomy section 1`] = `"<Fragment />"`;

exports[`XRef should render for PathologyOrBioTech section 1`] = `"<Fragment />"`;

exports[`XRef should render for ProteinProcessing section 1`] = `"<Fragment />"`;

exports[`XRef should render for Sequence section 1`] = `
"<Fragment>
  <XRefCategoryTable databaseCategory=\\"GENOME\\" databases={{...}} xRefData={{...}} accession=\\"P21802\\" />
</Fragment>"
`;

exports[`XRef should render for Structure section 1`] = `"<Fragment />"`;

exports[`XRef should render for SubCellularLocation section 1`] = `"<Fragment />"`;

exports[`XRefCategoryTable should render for database category GENOME and Ensembl 1`] = `
"<div>
  <hr />
  <h4>
    Genome annotation databases
  </h4>
  <XRefList database=\\"Ensembl\\" xRefData={{...}} accession=\\"P21802\\" />
</div>"
`;

exports[`XRefItem should render 1`] = `
"<Fragment>
  <Nr url=\\"https://www.ensembl.org/id/id value\\" newTab={true}>
    id value
  </Nr>
  description value
</Fragment>"
`;

exports[`XRefList should render for database Ensembl 1`] = `"<Component infoData={{...}} />"`;
